### ISSUE TITLE
Hunyuan Video 1.5

### ISSUE THREAD

**Issue by neph1** [tdrussell/diffusion-pipe#459]
Anyone knows what needs to be done? I can look into it, but my time is a bit constrained.

Edit: I'm just going to try it out and see what happens.

Edit2: Nope. Architecture has changed.

**Comment by tdrussell** [comment]
I will add this soon. Maybe not today, but hopefully within the next few days.

**Comment by neph1** [comment]
Sounds awesome. I started, got most of the submodule things rerouted, but got stuck on some things. I doubt I would save you much time, but let me know if you want to use it as a start.

**Comment by vAnNN47** [comment]
will there be an i2v support?

**Comment by neph1** [comment]
Hunyuan team updated their readme with some training instructions:

https://github.com/Tencent-Hunyuan/HunyuanVideo-1.5?tab=readme-ov-file#-training

> HunyuanVideo-1.5 is trained using the Muon optimizer, which accelerates convergence and improves training stability. The Muon optimizer combines momentum-based updates with Newton-Schulz orthogonalization for efficient optimization of large-scale video generation models.
> Creating a Muon Optimizer
> 
> Here's how to create a Muon optimizer for your model:
> 
> from hyvideo.optim.muon import get_muon_optimizer
> 
> # Create Muon optimizer for your model
> optimizer = get_muon_optimizer(
>     model=your_model,
>     lr=lr,                      # Learning rate
>     weight_decay=weight_decay,  # Weight decay
>     momentum=momentum,          # Momentum coefficient
>     adamw_betas=adamw_betas,   # AdamW betas for 1D parameters
>     adamw_eps=adamw_eps        # AdamW epsilon
> )
> 
>     📝 To be continued: More training details and the complete training pipeline will be released soon. Stay tuned!
>

**Comment by kabachuha** [comment]
@neph1 diffusion-pipe already has the Muon optimizer, btw

To use it, you need something like this

```
[optimizer]
type = 'genericoptim'
lr = 2e-5
muon = true
weight_decay = 0.015
```

also, muon can be changed to adamuon = true, to use a more recent technique

I added here a newer Muon-based optimizer, which has set a new record on the ModdedGPT speedrun - NorMuon, as well as the "cautious "weight decay technique from that repository, you can check them you at my pull request https://github.com/tdrussell/diffusion-pipe/pull/465

**Comment by neph1** [comment]
We might want to wait until they release the training code. I got the model loaded, but compared to 1.0 there seemed to be some pieces missing. But maybe others can get further.

**Comment by kabachuha** [comment]
@neph1 Simpletuner did an implementation, however

**Comment by mrreclusive3-beep** [comment]
training codes been out for 5 hours, whats the hold up man? lol.
im kidding, seriously, just kidding. just popped over here to see what's being talked about and if it was even on the radar.

i looked at the train.py myself, not at all coded for lora's and figuring that out, waaaay over my head.

@neph1 @tdrussell

**Comment by kabachuha** [comment]
Can't wait 😬

**Comment by mrreclusive3-beep** [comment]
> Can't wait 😬

i've spent the last 2 days trying to simpletuners implementation to work (ps, it doesn't)

Edit:
was able to get musubi tuner's implementation working though.

**Comment by kabachuha** [comment]
Official LoRA training script https://github.com/Tencent-Hunyuan/HunyuanVideo-1.5/commit/583711665c0ee6f48d163bd76e0557a10eb82752

**Comment by frauttauteffasu** [comment]
Initial HunyuanVideo1.5 supports looks to have been added in https://github.com/tdrussell/diffusion-pipe/commit/203e2c1507ad27b6f1ff7348db4715f2484db1a2 if anyone wants to test it.

**Comment by tdrussell** [comment]
It is not completely working yet. I fixed a couple of things and am doing a test training run. I will update the documentation when it is confirmed working.

**Comment by neph1** [comment]
Can confirm it's working, and it's blazingly fast!
I trained this (still undertrained lora) in about 30 minutes. It seems to be about halfway.

No lora:
<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/df95bcd7-91b6-443e-888d-369ad6f4eba5" />

Lora (50s scifi):
<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/c5a553cb-aae9-439d-94fe-78b764ddcc9b" />

But I'm not looking forward to training loras for each resolution. Hopefully they transfer ok.

I also hope I'll find some better settings for inference.
